.github/workflows/gitops_branch_cleanup.yaml

```diff
@@ -38,14 +38,22 @@ jobs:
 
         echo "branch=${{ github.event.ref }}" >> $GITHUB_OUTPUT
         echo "slug=$SLUG" >> $GITHUB_OUTPUT
+        
+        BRANCH_NAME="${{ github.event.ref }}"
+        if [[ "$BRANCH_NAME" == *"canary" ]]; then
+          DIRECTORY="canaries"
+        else
+          DIRECTORY="branches"
+        fi
+        echo "directory=$DIRECTORY" >> $GITHUB_ENV
 
     - name: Gitops cleanup
       uses: ovatu/actions/.github/actions/gitops-cleanup@main
       id: cleanup
       with:
         repository: ${{ inputs.gitops-repo }}
         repository-ssh-key: ${{ secrets.GITOPS_REPO_PRIVATE_KEY }}
-        directory: apps/${{ inputs.gitops-name }}/branches/${{ steps.variables.outputs.slug }}
+        directory: apps/${{ inputs.gitops-name }}/${{ env.directory }}/${{ steps.variables.outputs.slug }}
```

Comment:
This will delete /canaries/* directories when the branch is deleted, don't need a seperate action script